mono/utils/memfuncs.c

```diff
@@ -288,7 +289,33 @@ mono_determine_physical_ram_size (void)
 		return 134217728;
 	}
 
-	return page_size * num_pages;
+	memsize = page_size * num_pages;	/* Calculate physical memory size */
+
+	gint64 restricted_limit = getRestrictedPhysicalMemoryLimit();	/* Check for any cgroup limit */
+	if (restricted_limit != 0) {
+		gchar *heapHardLimit = getenv("DOTNET_GCHeapHardLimit");	/* See if user has set a limit */
+		errno = 0;
+		if (heapHardLimit != NULL) {
+			guint64 gcLimit = strtoull(heapHardLimit, NULL, 16);
+			if ((errno == 0) && (gcLimit != 0))
+				restricted_limit = (restricted_limit < gcLimit ? restricted_limit : (gint64) gcLimit);
+		} else {
+			gchar *heapHardLimitPct = getenv("DOTNET_GCHeapHardLimitPercent"); /* User % limit? */
+			if (heapHardLimitPct != NULL) {
+				int gcLimit = strtoll(heapHardLimitPct, NULL, 16);
+				if ((gcLimit > 0) && (gcLimit <= 100)) 
+					restricted_limit = (gcLimit * restricted_limit) / 100;
+				else
+					restricted_limit = (3 * restricted_limit) / 4;	/* Use 75% limit of container */
+                        } else {
```

Comment:
nit: fix indentation.

Reply:
I am not sure what's wrong with the indent. tab is set to 8.

Reply:
@nealef some of the lines are using spaces.  Files under src/mono/mono should use tabs (with tab width equal to 8 as you said)

Reply:
Fixed